sunkit_image/time_lag.py

```diff
@@ -0,0 +1,268 @@
+"""
+This module contains functions for calculating the cross-correlation and time
+lag between intensity cubes.
+"""
+import numpy as np
+
+import astropy.units as u
+
+__all__ = [
+    "cross_correlation",
+    "get_lags",
+    "max_cross_correlation",
+    "time_lag",
+]
+
+
+@u.quantity_input
+def get_lags(time: u.s):
+    """
+    Convert an array of evenly spaced times to an array of time lags
+    evenly spaced between ``-max(time)`` and ``max(time)``.
+    """
+    delta_t = np.diff(time)
+    if not np.allclose(delta_t, delta_t[0]):
+        raise ValueError("Times must be evenly sampled")
+    delta_t = delta_t.cumsum(axis=0)
+    return np.hstack([-delta_t[::-1], np.array([0]), delta_t])
+
+
+@u.quantity_input
+def cross_correlation(signal_a, signal_b, lags: u.s):
+    """
+    Compute cross-correlation between two signals, as a function of lag
+
+    By the convolution theorem the cross-correlation between two signals
+    can be computed as,
+
+    .. math::
+
+        \mathcal{C}_{AB}(\\tau) &= \mathcal{I}_A(t)\star\mathcal{I}_B(t) \\\\
+        &= \mathcal{I}_A(-t)\\ast\mathcal{I}_B(t) \\\\
+        &= \mathscr{F}^{-1}\{\mathscr{F}\{\mathcal{I}_A(-t)\}\mathscr{F}\{\mathcal{I}_B(t)\}\}
```

Comment:
The alternative to this spectacular backslash escaping effort is to prefix the docstring with `r"""`.

Reply:
What's so incomprehensible about `\\\\\\\\\\\\\`? 😆

Reply:
I'll add the `r` and clean up all of the `\`